crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -0,0 +1,381 @@
+		} else {
+			plain_chain_spec = output_path.clone();
+			output_path.pop();
+			if !output_path.exists() {
+				// Generate the output path if needed
+				create_dir_all(&output_path)?;
+			}
+		}
+		plain_chain_spec.set_extension("json");
+
+		// Locate binary, if it doesn't exist trigger build.
+		let mode: Profile = self.release.into();
+		let cwd = current_dir().unwrap_or(PathBuf::from("./"));
+		let binary_path = match binary_path(&mode.target_folder(&cwd), &cwd.join("node")) {
+			Ok(binary_path) => binary_path,
+			_ => {
+				cli.info(format!("No node was not found. The project will be built locally."))?;
+				cli.warning("NOTE: this may take some time...")?;
+				build_parachain(&cwd, None, &mode, None)?
+			},
+		};
+
+		// Generate plain spec.
+		spinner.set_message("Generating plain chain specification...");
+		let mut generated_files =
+			vec![format!("Specification and artifacts generated at: {}", &output_path.display())];
+		generate_plain_chain_spec(&binary_path, &plain_chain_spec, para_id, self.default_bootnode)?;
+		generated_files.push(format!(
+			"Plain text chain specification file generated at: {}",
+			plain_chain_spec.display()
+		));
+
+		// Customize spec based on input.
+		let relay = self.relay.unwrap_or(RelayChain::PaseoLocal).to_string();
+		replace_relay_spec(&plain_chain_spec, &relay, "rococo-local")?;
+		let chain_type = self.chain_type.unwrap_or(ChainType::Development).to_string();
+		replace_chain_type(&plain_chain_spec, &chain_type, "Local")?;
+		if self.protocol_id.is_some() {
+			let protocol_id = self.protocol_id.unwrap_or("template-local".to_string());
+			replace_protocol_id(&plain_chain_spec, &protocol_id, "template-local")?;
+		}
+
+		// Generate raw spec.
+		spinner.set_message("Generating raw chain specification...");
+		let raw_spec_name = plain_chain_spec
+			.file_name()
+			.and_then(|s| s.to_str())
+			.unwrap_or(DEFAULT_SPEC_NAME);
+		let raw_chain_spec =
+			generate_raw_chain_spec(&binary_path, &plain_chain_spec, &raw_spec_name)?;
+		generated_files.push(format!(
+			"Raw chain specification file generated at: {}",
+			raw_chain_spec.display()
+		));
+
+		// Generate genesis artifacts.
+		if self.genesis_code {
+			spinner.set_message("Generating genesis code...");
+			let wasm_file_name = format!("para-{}.wasm", para_id);
+			let wasm_file = export_wasm_file(&binary_path, &raw_chain_spec, &wasm_file_name)?;
+			generated_files.push(format!(
+				"WebAssembly runtime file exported at: {}",
+				wasm_file.display().to_string()
+			));
+		}
+
+		if self.genesis_state {
+			spinner.set_message("Generating genesis state...");
+			let genesis_file_name = format!("para-{}-genesis-state", para_id);
+			let genesis_state_file =
+				generate_genesis_state_file(&binary_path, &raw_chain_spec, &genesis_file_name)?;
+			generated_files.push(format!(
+				"Genesis State file exported at: {}",
+				genesis_state_file.display().to_string()
+			));
+		}
+
+		console::Term::stderr().clear_last_lines(1)?;
```

Comment:
```suggestion

```

Dont think we need this anymore due to the spinner. It essentially removes the line break under the intro for inconsistent UI.

Reply:
I included it to get rid of the last message from the spinner as once the generation is complete it doesn't really add much information.
But it is true than when building on release it removes the last break like and it doesn't look too good.

:+1: